### PR TITLE
Kadir - JS Exercise Week1

### PR DESCRIPTION
**Pull request by laperla11**

### CODE REVIEW

File: week-1/G-strings-methods/exercise.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
 // Start by creating a variable `message`
-
-console.log(message);
+var name = "Kadir";
+var nameLength = name.length;
+console.log(
+  "My name is " + name + " and my name is " + name.length + " characters long"
```
Comment (clems27) [inline week-1/G-strings-methods/exercise.js]: @laperla11 Nice work

Comment (laperla11) [reply week-1/G-strings-methods/exercise.js]: thanks for your comments.

File: week-1/I-floats/exercise.js [hunk 2]
```diff
@@ -1,2 +1,13 @@
 var numberOfStudents = 15;
 var numberOfMentors = 8;
+var total = numberOfMentors + numberOfStudents;
+var studentPercentage = numberOfStudents / total;
+var studentRoughPercentage = Math.round(100 * studentPercentage);
+var mentorPercentage = numberOfMentors / total;
+var mentorRoughPercentage = Math.round(100 * mentorPercentage);
+
+var message1 = "Percentage students : " + studentRoughPercentage + "%";
```
Comment (Mussie14) [inline week-1/I-floats/exercise.js]: good job

**Comment by khan-zareen** [comment]
very nice job

File: week-1/I-floats/exercise.js (hunk 2 above)
Comment (laperla11) [reply week-1/I-floats/exercise.js]: Thanks Mussie.

File: week-1-practise/level-1/7-logic-error-4.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // The syntax for this function is valid but it has an error, find it and fix it.
 
 function multiply(a, b, c) {
-  a * b * c;
+  return a * b * c;
   return;
```
Comment (foaly-nr1) [inline week-1-practise/level-1/7-logic-error-4.js]: There's a line of code here that will never be executed. Can you spot it?

File: week-1-practise/level-1/8-function-output-1.js [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 
 // Answer these questions:
-// - What does this program do?
-// - How many parameters does the function take?
-// - What is the function name?
-// - Where is the function called? with what parameters?
\ No newline at end of file
+// - What does this program do? first pick up a number between 0 and 1 randomly and after multiply it with 10.
+// - How many parameters does the function take? this function doesn`t need any parameters.
+// - What is the function name? function name is getNumber().
```
Comment (foaly-nr1) [inline week-1-practise/level-1/8-function-output-1.js]: Very close but not quite. It's just a subtlety but I suggest reading the first few paragraphs here: http://2ality.com/2015/09/function-names-es6.html.

Comment (foaly-nr1) [reply week-1-practise/level-1/8-function-output-1.js]: You've got it right in `9-function-output-2.js` 😀

File: week-1-practise/level-2/4-currency-conversion-2.js [hunk 5]
```diff
@@ -7,8 +7,18 @@
...
-function convertToBRL(price) {}
+function convertToUSD(price) {
+  var result = price * 1.4;
+  var conversionFee = result * 0.01;
+  result += conversionFee;
+  return result.toFixed(2);
+}
+function convertToBRL(price) {
+  var result = price * 4.93;
+  var conversionFee = result * 0.01;
```
Comment (foaly-nr1) [inline week-1-practise/level-2/4-currency-conversion-2.js]: Can you think of a way to apply the DRY principle (don't repeat yourself) here? This is a very good example of code that can be improved through DRY – you have 2 identical lines of code!

File: week-1/E-strings/exercise.js [hunk 6]
```diff
@@ -1,3 +1,5 @@
 // Start by creating a variable `message`
+var array = [1, 2, 3, 4];
```
Comment (foaly-nr1) [inline week-1/E-strings/exercise.js]: I'm not sure `array` is the best name for an array. Can you think of a more meaningful one?

File: week-1/I-floats/exercise.js [hunk 7]
```diff
@@ -1,2 +1,13 @@
 var numberOfStudents = 15;
 var numberOfMentors = 8;
+var total = numberOfMentors + numberOfStudents;
+var studentPercentage = numberOfStudents / total;
+var studentRoughPercentage = Math.round(100 * studentPercentage);
+var mentorPercentage = numberOfMentors / total;
+var mentorRoughPercentage = Math.round(100 * mentorPercentage);
```
Comment (foaly-nr1) [inline week-1/I-floats/exercise.js]: Can you think of a way to use a function to make this more DRY (don't repeat yourself)?

File: week-1/L-functions-nested/exercise.js [hunk 8]
```diff
@@ -1,5 +1,22 @@
...
-var mentor4 = "Rob";
-var mentor5 = "Yohannes";
+function percentage(number) {
+  return Math.round(number);
+}
+
+function message(numberOfStudents, numberOfMentors) {
+  var total = numberOfStudents + numberOfMentors;
+  var studentsPercentage = (100 * numberOfStudents) / total;
+  var mentorsPercentage = (100 * numberOfMentors) / total;
```
Comment (foaly-nr1) [inline week-1/L-functions-nested/exercise.js]: Similarly to your code in `I-floats/exercise.js`, can you use another function to reduce duplicate code?

File: week-1/L-functions-nested/exercise2.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+var mentor2 = "Irina";
+var mentor3 = "Mimi";
+var mentor4 = "Rob";
+var mentor5 = "Yohannes";
+
+function uppercase(name) {
+  return name.toUpperCase();
+}
+
+function shoutyGreeting(name) {
```
Comment (foaly-nr1) [inline week-1/L-functions-nested/exercise2.js]: Really good work on giving the functions "speaking" names! 😀👏🏻

Comment (laperla11) [reply week-1/L-functions-nested/exercise2.js]: Thanks.

File: week-1/L-functions-nested/exercise.js (hunk 8 above)
Comment (laperla11) [reply week-1/L-functions-nested/exercise.js]: I will use a function in `I-floats/exercise.js` as well.

File: week-1/I-floats/exercise.js (hunk 7 above)
Comment (laperla11) [reply week-1/I-floats/exercise.js]: yes I will do.

File: week-1/E-strings/exercise.js (hunk 6 above)
Comment (laperla11) [reply week-1/E-strings/exercise.js]: I`ve changed the array name as `arrayExample`.

Comment (laperla11) [reply week-1/E-strings/exercise.js]: I have changed the array name as `arrayExample`.

File: week-1/I-floats/exercise.js (hunk 2 above)
Comment (ZaimAbz) [reply week-1/I-floats/exercise.js]: nice job I had to comment as well, hopefully next time I will give a useful one lol

File: week-1/E-strings/exercise.js (hunk 6 above)
Comment (foaly-nr1) [reply week-1/E-strings/exercise.js]: Hm how about something along the lines of `number`? The name of a variable should always describe its value (what's in it).

File: week-1-practise/level-1/7-logic-error-4.js (hunk 3 above)
Comment (laperla11) [reply week-1-practise/level-1/7-logic-error-4.js]: thanks, of course `return;`

File: week-1-practise/level-2/4-currency-conversion-2.js (hunk 5 above)
Comment (laperla11) [reply week-1-practise/level-2/4-currency-conversion-2.js]: Thanks for the reminder, DRY implemented.

File: week-1/E-strings/exercise.js (hunk 6 above)
Comment (laperla11) [reply week-1/E-strings/exercise.js]: `arrayNumbers` looks more related with the content of the array.

File: week-1/I-floats/exercise.js (hunk 7 above)
Comment (laperla11) [reply week-1/I-floats/exercise.js]: DRY implemented by using a function for displaying the message on the screen and a nested function for rounding the percentage.

File: week-1/L-functions-nested/exercise.js (hunk 8 above)
Comment (laperla11) [reply week-1/L-functions-nested/exercise.js]: DRY implemented by using a function for displaying the message on the screen and a nested function for rounding the percentage.

Comment (laperla11) [reply week-1/L-functions-nested/exercise.js]: also duplicated code removed.

File: week-1-practise/level-2/4-currency-conversion-2.js [hunk 10]
```diff
@@ -7,8 +7,18 @@
   Find a way to add 1% to all currency conversions (think about the DRY principle)
 */
 
-function convertToUSD(price) {}
-function convertToBRL(price) {}
+function convertToUSD(price) {
+  var result = price * 1.4;
+  var conversionFee = result * 0.01;
```
[the remark this answers is not in the document]
Comment (laperla11) [reply week-1-practise/level-2/4-currency-conversion-2.js]: Ahmet, thanks for your contribution. I have already included the function.

File: week-1/I-floats/exercise.js (hunk 2 above)
Comment (laperla11) [reply week-1/I-floats/exercise.js]: Abz, thanks for you kind words.

Comment (Ninaa117) [reply week-1/I-floats/exercise.js]: easy to understand good work :)

Comment (laperla11) [reply week-1/I-floats/exercise.js]: Thank you very much, Nina.

**Comment by laperla11** [comment]
> very nice job

Thanks Zareen.

**Pull request closed without merging**